pgvectorscale/src/access_method/upgrade_test.rs

```diff
@@ -194,4 +193,193 @@ pub mod tests {
         let cnt: i64 = client.query_one(&format!("WITH cte as (select * from test order by embedding <=> '[1,1,1,{suffix}]') SELECT count(*) from cte;"), &[]).unwrap().get(0);
         assert_eq!(cnt, 303, "count after upgrade");
     }
+
+    #[test]
+    fn test_upgrade_from_0_2_0() {
+        test_upgrade_base("0.2.0", "0.11.4");
+    }
+
+    #[test]
+    fn test_upgrade_from_0_3_0() {
+        test_upgrade_base("0.3.0", "0.11.4");
+    }
+
+    #[test]
+    fn test_upgrade_from_0_4_0() {
+        test_upgrade_base("0.4.0", "0.12.5");
+    }
+
+    /// Common upgrade test logic for versions 0.2.0, 0.3.0, and 0.4.0
+    fn test_upgrade_base(version: &str, pgrx_version: &str) {
+        if cfg!(feature = "pg17") {
+            // PG17 is only supported for one version
```

Comment:
This doesn't need to be fixed in this PR, but we should support update tests on PG17 now that we've released a version that supports it.